src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsHostBuilderExtensions.cs

```diff
@@ -57,52 +60,57 @@ public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, str
+                    excludedDomains.Add("core.usgovcloudapi.net");
+                    excludedDomains.Add("localhost");
+                    excludedDomains.Add("127.0.0.1");
+
+                    return dependencyCollector;
+                });
 
                 ServerTelemetryChannel serverChannel = new ServerTelemetryChannel();
-                services.AddSingleton<ITelemetryModule>(serverChannel);
                 services.AddSingleton<ITelemetryChannel>(serverChannel);
                 services.AddSingleton<TelemetryConfiguration>(provider =>
                 {
                     ITelemetryChannel channel = provider.GetService<ITelemetryChannel>();
-                    TelemetryConfiguration config = new TelemetryConfiguration(instrumentationKey, channel);
-                    
-                    foreach (ITelemetryInitializer initializer in provider.GetServices<ITelemetryInitializer>())
-                    {
-                        config.TelemetryInitializers.Add(initializer);
-                    }
-
-                    QuickPulseTelemetryModule quickPulseModule = null;
-                    foreach (ITelemetryModule module in provider.GetServices<ITelemetryModule>())
-                    {
-                        if (module is QuickPulseTelemetryModule telemetryModule)
-                        {
-                            quickPulseModule = telemetryModule;
-                        }
-                        module.Initialize(config);
-                    }
-
-                    QuickPulseTelemetryProcessor processor = null;
-                    config.TelemetryProcessorChainBuilder
-                        .Use((next) =>
-                        {
-                            processor = new QuickPulseTelemetryProcessor(next);
-                            return processor;
-                        })
-                        .Use((next) => new FilteringTelemetryProcessor(filter, next));
+                    TelemetryConfiguration config = TelemetryConfiguration.CreateDefault();
+                    SetupTelemetryConfiguration(
+                        config,
+                        instrumentationKey, 
+                        filter, 
+                        samplingSettings,
+                        channel,
+                        provider.GetServices<ITelemetryInitializer>(),
+                        provider.GetServices<ITelemetryModule>());
 
-                    if (samplingSettings != null)
+                    // Function users have no access to TelemetryConfiguration from host DI container,
+                    // so we'll expect user to work with TelemteryConfiguration.Active
+                    // Also, some ApplicaitonInsights internal operations (heartbeats) depend on
+                    // the TelemetryConfiguration.Active being set so, we'll set up Active once per process lifetime.
+                    TelemetryConfiguration activeConfig = TelemetryConfiguration.Active;
```

Comment:
Customers could change this config from within their code, right? Would that affect anything critical?

Reply:
1. WebJobs SDK customers:
We'll need to update documentation and encourage them to modify telemetry configuration through the dependency injection. They can do all kinds of harmful things  - remove initializers, processors and break things.
The  good thing - we can refer to existing ApplicationInsights documentation on how to do it correctly without breaking things.

2.  Azure Function customers:
 They do not have access to DI and only have access to TelemetryConfiguration.Active
They may change Active config, but it would not affect the instance host uses for logging. I.e. changes would only reflect telemetry that is reported directly from the user code.
Active is needed to just support the scenario when telemetry is reported from the user code rather than modifications.
I think the only appropriate way to modify host configuration is still through the settings file and is quite limited for functions users.

Reply:
Great, this all makes sense. It's still a step in the right direction to allow TelemetryConfiguration.Active have the same setup as our client uses. 

Another option I'd considered was having a TelemetryClient binding where we'd pass you the one we use in DI directly.

Reply:
> Another option I'd considered was having a TelemetryClient binding where we'd pass you the one we use in DI directly.

This would be awesome!